plugin/servergen.go

```diff
@@ -420,28 +454,61 @@ func (p *OrmPlugin) getMethodProps(method *descriptor.MethodDescriptorProto) (ge
 func (p *OrmPlugin) generatePreserviceCall(svc, typeName, mthd string) {
 	p.P(`if custom, ok := interface{}(in).(`, svc, typeName, `WithBefore`, mthd, `); ok {`)
 	p.P(`var err error`)
-	p.P(`ctx, db, err = custom.Before`, mthd, `(ctx, in, db)`)
+	p.P(`db, err = custom.Before`, mthd, `(ctx, db)`)
 	p.P(`if err != nil {`)
 	p.P(`return nil, err`)
 	p.P(`}`)
 	p.P(`}`)
 }
 
-func (p *OrmPlugin) generatePreserviceHook(svc, typeName, inTypeName, mthd string) {
+func (p *OrmPlugin) generatePreserviceHook(svc, typeName, mthd string) {
 	p.P(`// `, svc, typeName, `WithBefore`, mthd, ` called before Default`, mthd, typeName, ` in the default `, mthd, ` handler`)
 	p.P(`type `, svc, typeName, `WithBefore`, mthd, ` interface {`)
-	p.P(`Before`, mthd, `(context.Context, *`, inTypeName, `, *`, p.Import(gormImport), `.DB) (context.Context, *`, p.Import(gormImport), `.DB, error)`)
+	p.P(`Before`, mthd, `(context.Context, *`, p.Import(gormImport), `.DB) (*`, p.Import(gormImport), `.DB, error)`)
+	p.P(`}`)
+}
+
+func (p *OrmPlugin) generatePostserviceCall(svc, typeName, mthd string) {
+	p.P(`if custom, ok := interface{}(in).(`, svc, typeName, `WithAfter`, mthd, `); ok {`)
+	p.P(`var err error`)
+	p.P(`err = custom.After`, mthd, `(ctx, out, db)`)
+	p.P(`if err != nil {`)
+	p.P(`return nil, err`)
 	p.P(`}`)
+	p.P(`}`)
+}
+
+func (p *OrmPlugin) generatePostserviceHook(svc, typeName, outTypeName, mthd string) {
+	p.P(`// `, svc, typeName, `WithAfter`, mthd, ` called before Default`, mthd, typeName, ` in the default `, mthd, ` handler`)
+	p.P(`type `, svc, typeName, `WithAfter`, mthd, ` interface {`)
+	p.P(`After`, mthd, `(context.Context, *`, outTypeName, `, *`, p.Import(gormImport), `.DB) error`)
+	p.P(`}`)
+}
+
+func (p *OrmPlugin) hasFieldSelection(object generator.Object) string {
+	return p.hasFieldOfType(object, "FieldSelection")
+}
+
+func (p *OrmPlugin) hasFiltering(object generator.Object) string {
+	return p.hasFieldOfType(object, "Filtering")
+}
+
+func (p *OrmPlugin) hasSorting(object generator.Object) string {
+	return p.hasFieldOfType(object, "Sorting")
+}
+
+func (p *OrmPlugin) hasPagination(object generator.Object) string {
+	return p.hasFieldOfType(object, "Pagination")
 }
 
-func (p *OrmPlugin) hasFieldsSelector(object generator.Object) string {
+func (p *OrmPlugin) hasFieldOfType(object generator.Object, fieldType string) string {
```

Comment:
I'm the one who first made this naming choice `has...`, but since it returns the field name, `get...` probably makes more sense.